typescript/sdk/src/core/read.ts

```diff
@@ -0,0 +1,54 @@
+import { providers } from 'ethers';
+
+import { Mailbox__factory } from '@hyperlane-xyz/core';
+import { Address, objMap, promiseObjAll } from '@hyperlane-xyz/utils';
+
+import { EvmHookReader } from '../hook/read.js';
+import { EvmIsmReader } from '../ism/read.js';
+import { MultiProvider } from '../providers/MultiProvider.js';
+import { ChainName } from '../types.js';
+
+import { CoreConfig } from './types.js';
+
```

Comment:
can you make `EvmCoreReader` implement a `CoreReader` interface with `deriveCoreConfig` exposed - this is so I can keep the reading logic in the abstract crud modules instead of a VM-specfic implementation

Reply:
Done